src/Microsoft.Identity.Client/Config/ConfidentialApplicationOptions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Identity.Client.Config
+{
+    /// <summary>
+    /// </summary>
+    public class ConfidentialApplicationOptions
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        public string ClientCredentialSecret { get; set; }
```

Comment:
We should talk about this.  Since the config system uses types to define the json parsing in the options files, we currently have a block of "ApplicationOptions" (for common stuff) and a ConfidentialApplicationOptions (that you'd add with .WithConfidentialApplicationOptions()).  I can see the derivation case as well, but then we'd want a (currently empty) PublicClientApplicationOptions that derives as well and then modify the static create methods.  I'd love feedback here on desired API surface.

---
In reply to: [241386267](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/723#discussion_r241386267) [](ancestors = 241386267)